starter-bots/scala/src/main/scala/za/co/entelect/challenge/Bot.scala

```diff
@@ -0,0 +1,131 @@
+package za.co.entelect.challenge
+
+import za.co.entelect.challenge.commands.{Command, DigCommand, DoNothingCommand, MoveCommand, ShootCommand}
+import za.co.entelect.challenge.entities.{Cell, GameState, MyWorm, Opponent, Position, Worm}
+import za.co.entelect.challenge.enums.CellType.CellTypes
+import za.co.entelect.challenge.enums.{CellType, Direction}
+import za.co.entelect.challenge.enums.Direction.{Direction, DirectionObject, Directions}
+
+import scala.collection.mutable.ListBuffer
+import scala.util.Random
+
+class Bot(gameState: GameState, random: Random){
+  private val opponent: Opponent  = gameState.opponents.head
+  private val currentWorm: MyWorm = getCurrentWorm(gameState)
+
+  private def euclideanDistance(aX: Int, aY: Int, bX: Int, bY: Int): Int = {
+    Math.sqrt(
+      Math.pow(aX - bX, 2) + Math.pow(aY - bY, 2)
+    ).toInt
+  }
+
+  private def isValidCoordinate(x: Int, y: Int): Boolean = {
+    x >= 0 && x < gameState.mapSize && y >= 0 && y < gameState.mapSize
+  }
+
+  private def getCurrentWorm(gameState: GameState): MyWorm = {
+    gameState.myPlayer.worms
+      .filter(myWorm => myWorm.id == gameState.currentWormId)
+      .head
+  }
+
+  private def resolveDirection(a: Position, b: Position): Direction = {
+    val verticalComponent = b.y - a.y
+    val horizontalComponent = b.x - a.x
+
+    val builder: StringBuilder = new StringBuilder()
+
+    if     (verticalComponent < 0) builder.append('N')
+    else if(verticalComponent > 0) builder.append('S')
+
+    if     (horizontalComponent < 0) builder.append('W')
+    else if(horizontalComponent > 0) builder.append('E')
+
+    Directions.withNameInsensitive(builder.toString())
+  }
+
+  private def getSurroundingCells(x: Int, y: Int): List[Cell] = {
+    val cells: ListBuffer[Cell] = ListBuffer.empty[Cell]
+
+    for (i <- Range(x - 1, x + 2)) {
+      for (j <- Range(y - 1, y + 2)) {
```

Comment:
Consider putting both of these into one `for` statement to increase readability, eg:
```scala
for (
  i <- Range(x - 1, x + 2);
  j <- Range(y - 1, y + 2)
) {
```